compiler/rustc_mir_transform/src/shim.rs

```diff
@@ -647,7 +647,7 @@ fn build_call_shim<'tcx>(
     let mut sig = if let Some(sig_substs) = sig_substs {
         sig.subst(tcx, &sig_substs)
     } else {
-        sig.skip_binder()
+        sig.subst_identity()
```

Comment:
I remember a similar pattern in `Instance::subst_mir` that I used `skip_binder` for previously:

https://github.com/rust-lang/rust/blob/a9baa16482ba4100df529ccba39c787f27ad0475/compiler/rustc_middle/src/ty/instance.rs#L581-L591

Should that one also be `subst_identity()`?

Reply:
Yes, probably